### PR TITLE
test(mcp): prove the MCP tool-call rule surface end to end

### PR DESCRIPTION
**Pull request by irfndi**
> **Stacked on #48** (base is `issue/2-block-tool-call-before-run`, itself stacked on #45 → #38). Merge that chain first; GitHub then retargets this PR to `master`. Diff shows only this issue's work.

## What changed

- **`preToolUse.test.ts`** — new MCP suite: a `mcp__postgres__*` rule judges `mcp__postgres__query` on its `tool_input` while a **contradicting `tool_response`** sits in the payload, both directions — dangerous args + harmless output → block; clean args + alarming output → silent, with the captured judge state proving only the arguments arrived. Act blocks with `Rule "no-drop-through-postgres"` in the reason; flag lets the call run with the note; ignore passes silently. Out-of-scope calls (`mcp__slack__postMessage`, `Bash`) stay silent and un-judged; failed and timed-out checks let the MCP call run with `CHECK_FAILED`/`CHECK_TIMEOUT` logged.
- **`scope.test.ts`** — server-boundary cases: `mcp__postgres__*` does not match `mcp__postgres2__query`, `mcp__puppeteer__screenshot` or `mcp__postgres-inspector__query`, while `mcp__postgres-inspector__*` matches `mcp__postgres-inspector__validate_migration` (names taken from #3's payloads and this repo's fixtures).
- **`hook.test.ts`** — real-binary e2e: `pre-tool-use` for `mcp__postgres__query` without a key exits 0, prints nothing, and logs the skip naming the tool.
- **`skills/oh-my-plumb-compile/SKILL.md`** — authoring guidance under question-writing: `target: "toolCall"` + tool-name scopes + `when: "edit"`, a worked DROP/TRUNCATE rule, and the standing rules (ask about command and arguments, never output or conversation; flag by default, act only for absolute wording) covering all three examples from the issue — never DROP/TRUNCATE, read-only production, don't post to Slack without asking.

## Why

Kilo flagged #11 as a 90% duplicate of #2. This implements the remaining MCP-specific surface on top of #2's generic pre-execution block: prove the capability end to end with real MCP naming and argument-vs-output payloads, and give rule authors the MCP guidance the skill lacked. Zero src changes — the generic machinery from the parent chain already covers it, and these tests pin that down (the issue's three acceptance bullets are all assertions now, not hopes).

## Verification

- `pnpm format:check` — pass
- `pnpm -r run build` — pass
- `pnpm -r run typecheck` — pass
- `pnpm -r run test` — pass (schema 7 tests/1 file; cli 139 tests/26 files including the 7 new MCP tests; 146 total)

Depends on: #1, #2, #3 (chain `#38 → #45 → #48`)

Closes #11

## Summary by Sourcery

Prove the MCP tool-call rule surface end to end and document how to author safe, argument-based MCP rules.

Enhancements:
- Validate MCP tool-call rule behavior end to end, including argument-only judging, verdict handling, failure recovery, and precise server-scoped matching.

Documentation:
- Document how to author tool-call rules for MCP and other tools, including scope selection and argument-focused questions.

Tests:
- Add end-to-end coverage for MCP pre-tool-use behavior, scope boundaries, skipped calls without credentials, and failed or timed-out checks.

### CODE REVIEW

**Comment by irfndi** [comment]
Base restored to `issue/2-block-tool-call-before-run`: this PR's tests exercise the `pre-tool-use` hook route that only exists in #48, so it is correctly stacked rather than conflicting. After #48 merges, retarget to master (same pattern as the rest of the batch) — the diff then collapses to this PR's own 4 files.

File: packages/cli/test/preToolUse.test.ts [hunk 1]
```diff
@@ -16,14 +16,17 @@ import { emit } from "../src/lib/output.js";
...
+const model = vi.hoisted(
+  (): { answer?: Record<string, unknown>; failure?: Error; seen?: string } => ({}),
+);
 
 vi.mock("ai", async (importOriginal) => {
   const actual = await importOriginal<typeof AiModule>();
   return {
     ...actual,
-    experimental_evaluate: async () => {
+    experimental_evaluate: async (options: { state: unknown }) => {
```
[the remark this answers is not in the document]
Comment (irfndi) [reply packages/cli/test/preToolUse.test.ts]: Fixed: the mock throws 'evaluate was called without a state' if state is ever missing, so a refactor fails with that message instead of an undefined read.

File: packages/cli/test/preToolUse.test.ts [hunk 2]
```diff
@@ -152,6 +167,94 @@ describe("blocking a recorded call before it runs", () => {
...
 });
 
+describe("a rule scoped to one MCP server", () => {
+  const mcpPayload = (cwd: string, tool_input: Record<string, unknown>) =>
+    payload(cwd, "mcp__postgres__query", tool_input);
+
+  it("judges the call on its arguments even when the output says the opposite", async () => {
+    violation(0.97, "no-drop-through-postgres");
+    const root = repoWith([mcpRule]);
+    const out = await handlePostToolUse({
```
[the remark this answers is not in the document]
Comment (irfndi) [reply packages/cli/test/preToolUse.test.ts]: After #48 merged, the post-hook no longer judges tool calls at all; it only logs them. These tests now assert exactly that: the post-hook returns silent and the judge is never called after the run, so a contradicting tool_response cannot reach it. The comments in the tests say so.

File: skills/oh-my-plumb-compile/SKILL.md [hunk 3]
```diff
@@ -56,11 +56,42 @@ The judge is a small, fast model that answers typed questions with a probability
...
   - `choice` when the rule names a closed set of shapes and only some are wrong: `criteria` maps each option to a description, `violating` lists the wrong ones. Example: a service must be a class with static methods, so options are `class-static`, `loose-functions`, `eager-init` and `violating` is the last two.
   - `score` when the rule is a matter of degree: `criteria` is an ordered list of levels from compliant (index 0) to worst, and `violatingFrom` is the first level that counts as broken. Example for "minimum code that solves the problem": levels "as short as it can be", "somewhat longer than needed", "about twice as long", "several times longer", with `violatingFrom: 2`.
 
+### A rule about a tool call, not a change
+
+Some rules judge what a call does rather than what a diff looks like: "never `DROP` or `TRUNCATE` through the postgres MCP", "read-only queries in production", "don't post to Slack without asking". For those set `"target": "toolCall"`, put tool names in `scope` instead of file globs, and always use `"when": "edit"` — the judge runs the moment the call arrives, before it runs, and stops it only in the act band.
```
[the remark this answers is not in the document]
Comment (irfndi) [reply skills/oh-my-plumb-compile/SKILL.md]: This changed underneath: #46 gave turn-phase toolCall rules a path (judged at Stop on the turn's tool-call log), so a when: turn tool-call rule does run now. The skill section here now tells the author to use edit for a rule about one call, and points to the turn kind for rules about which tools a turn used.
